src/CoreShop/Bundle/FrontendBundle/Controller/CategoryController.php

```diff
@@ -87,18 +96,19 @@ public function indexAction(Request $request)
 
         $perPage = $request->get('perPage', $defaultPerPage);
 
-        $category = $this->getRepository()->find($request->get('category'));
-
-        if (!$category instanceof CategoryInterface) {
-            return $this->redirectToRoute('coreshop_index');
+        $categoryList = $this->getRepository()->findBy([$this->repositoryIdentifier => $request->get($this->requestIdentifier)], null, 1);
```

Comment:
not quite happy with that, maybe we could add a findUniqueBy function to the pimcore class?

Reply:
done. findOneBy was already implemented.